shared/middleware/helpers/bucket_helper.rb

```diff
@@ -338,6 +338,13 @@ def hard_delete_channel_content(encrypted_channel_id)
     result.deleted.count
   end
 
+  # Check whether a project has any S3 Objects in this bucket.
+  def project_contains_objects?(storage_id, storage_app_id)
```

Comment:
Maybe a comment about what storage_id and storage_app_id are? Or point to the models they come from? Some context would be nice because its a little confusing that the method is called `project_contains_objects?` yet none of that parameters obviously reference a "project".